### ISSUE TITLE
Starters do not depend on `spring-boot-starter`

### ISSUE THREAD

**Issue by wilkinsona** [spring-cloud/spring-cloud-zookeeper#230]
As [noted in the documentation](https://docs.spring.io/spring-boot/docs/2.1.6.RELEASE/reference/htmlsingle/#boot-features-custom-starter-module-starter) starters are required to directly or indirectly depend upon `spring-boot-starter`. The starters in this repository do not meet this requirement so when they are using in isolation, things like `@SpringBootApplication` are not on the classpath.

**Comment by spencergibb** [comment]
Does it make sense to move this to spring cloud commons so there doesn't have to be an issue in every repo?

**Comment by wilkinsona** [comment]
Sorry, I'd assumed it was an isolated problem based on https://github.com/spring-io/start.spring.io/issues/191. I thought we would have heard about it before if it applies to the Cloud starters that, I imagine, are more commonly used. I guess it's unusual to use one in isolation though? 

If the problem's more widespread then, yeah, it probably makes sense to handle in centrally than have loads of issues. Whatever works best for you.

**Comment by spencergibb** [comment]
I'll investigate. The starters here depends on commons directly rather that spring cloud starter, that may be enough

**Comment by wilkinsona** [comment]
FWIW, `spring-cloud-commons` seems to be OK as `spring-cloud-starter` has the required dependency. `spring-cloud-starter-config` is fine too as it depends on `spring-cloud-starter`.

**Comment by spencergibb** [comment]
Yeah, not in front of computer to check efficiently, so it is likely localized.

**Comment by mbhave** [comment]
`spring-cloud-starter-consul-config` appears to be broken as well so maybe it's a broader problem. Projects generated with `spring-cloud-starter-vault-config` aren't broken as we explicitly mark that one as [not a starter](https://github.com/spring-io/start.spring.io/blob/3fd8e39d730070f0b4548f15efeaaca804dbe312/start-site/src/main/resources/application.yml#L909).

**Comment by snicoll** [comment]
> Projects generated with spring-cloud-starter-vault-config aren't broken as we explicitly mark that one as not a starter.

It was a workaround until the definition of the starter got fixed. I thought I had created an issue back then but it looks like I forgot.

**Comment by wilkinsona** [comment]
Mark fixed the vault one a while ago so we needed to catch up. See https://github.com/spring-io/start.spring.io/issues/193.

**Comment by snicoll** [comment]
I was about to write that, see https://github.com/spring-io/initializr/issues/414#issuecomment-296957836

**Comment by wilkinsona** [comment]
I've in the process of writing some tests for start.spring.io to check that a dependency that is named like a starter actually is one. I've identified the following modules that should depend on `spring-boot-starter` but do not do so:

- `spring-cloud-starter-aws`
- `spring-cloud-starter-aws-jdbc`
- `spring-cloud-starter-aws-messaging`
- `spring-cloud-starter-consul-config`
- `spring-cloud-starter-zookeeper-config`

There may be others as the start.spring.io tests only cover the starters that are offered by the service.

`spring-cloud-services-starter-config-client` is also affected but I'll open a separate issue for that.
